src/fsverity/ioctl.rs

```diff
@@ -71,8 +75,33 @@ pub fn fs_ioc_measure_verity<F: AsFd, H: FsVerityHashValue>(fd: F) -> Result<H>
     }
 
     if digest.digest_algorithm != digest_algorithm || digest.digest_size != digest_size {
-        Err(std::io::Error::from(std::io::ErrorKind::InvalidData))?
+        Err(Errno::ILSEQ)
```

Comment:
If we do that then we need to also decide what to do about `EOVERFLOW`.  According to the docs, https://docs.kernel.org/filesystems/fsverity.html#fs-ioc-measure-verity we get `EOVERFLOW` if the digest is longer than specified in the digest_size.  So if we expect sha256 (which we do) but the file has sha512, then we'll fail at the syscall level with `EOVERFLOW` and never make it to this point.

Potential ways forward:
 - always use a larger buffer
 - return `EOVERFLOW` also for digest mismatch detected here
 - detect the `EOVERFLOW` case and turn it into `EILSEQ`.

I personally find `EILSEQ` a fairly weird choice, but `EOVERFLOW` is also sort of strange.  Anything but `EINVAL` :sweat_smile: 

But what about #25?  Maybe this would be a good chance to look into that.  This is Rust, not C: we can do better...

Reply:
> But what about https://github.com/containers/composefs-rs/issues/25? Maybe this would be a good chance to look into that. This is Rust, not C: we can do better...

I started looking at it (I can push up my WIP branch), but it's quite an infectious thing, although I tried starting from a different place.

I'll take a crack at updating this PR instead and target it at just this code.